src/ruby/ext/grpc/rb_compression_options.c

```diff
@@ -0,0 +1,592 @@
+                   grpc_rb_compression_options_init, -1);
+
+  /* Gets the bitset of enabled algorithms. */
+  rb_define_method(grpc_rb_cCompressionOptions, "enabled_algorithms_bitset",
+                   grpc_rb_compression_options_get_enabled_algorithms_bitset,
+                   0);
+
+  /* Methods for getting the default algorithm, default level, and disabled
+   * algorithms as readable names. */
+  rb_define_method(grpc_rb_cCompressionOptions, "default_algorithm",
+                   grpc_rb_compression_options_get_default_algorithm, 0);
+  rb_define_method(grpc_rb_cCompressionOptions, "default_level",
+                   grpc_rb_compression_options_get_default_level, 0);
+  rb_define_method(grpc_rb_cCompressionOptions, "disabled_algorithms",
+                   grpc_rb_compression_options_get_disabled_algorithms, 0);
+
+  /* Methods for getting the internal enum default algorithm and level enum
+   * values of an instance. */
+  rb_define_method(
+      grpc_rb_cCompressionOptions, "default_algorithm_internal_value",
+      grpc_rb_compression_options_get_default_algorithm_internal_value, 0);
+  rb_define_method(grpc_rb_cCompressionOptions, "default_level_internal_value",
+                   grpc_rb_compression_options_get_default_level_internal_value,
+                   0);
+
+  /* Determines whether or not an algorithm is enabled, given a readable
+   * algorithm name.*/
+  rb_define_method(grpc_rb_cCompressionOptions, "is_algorithm_enabled",
+                   grpc_rb_compression_options_is_algorithm_enabled, 1);
+
+  /* Methods for converting to and from algorithm enum values and their readable
+   * names. */
+  rb_define_singleton_method(
+      grpc_rb_cCompressionOptions, "algorithm_name_to_value",
+      grpc_rb_compression_options_algorithm_name_to_value, 1);
+  rb_define_singleton_method(
+      grpc_rb_cCompressionOptions, "algorithm_value_to_name",
+      grpc_rb_compression_options_algorithm_value_to_name, 1);
+
+  /* Methods for converting to and from compression level enum values and their
+   * readable names. */
+  rb_define_singleton_method(grpc_rb_cCompressionOptions, "level_name_to_value",
+                             grpc_rb_compression_options_level_name_to_value,
+                             1);
+  rb_define_singleton_method(grpc_rb_cCompressionOptions, "level_value_to_name",
+                             grpc_rb_compression_options_level_value_to_name,
+                             1);
+
+  /* Provides a hash of the compression settings suitable
+   * for passing to server or channel args. */
+  rb_define_method(grpc_rb_cCompressionOptions, "to_hash",
+                   grpc_rb_compression_options_to_hash, 0);
+  rb_define_alias(grpc_rb_cCompressionOptions, "to_channel_arg_hash",
+                  "to_hash");
+
+  /* Ruby symbols for the names of the different compression levels. */
+  rb_define_const(grpc_rb_cCompressionOptions, "COMPRESS_NONE_SYM",
+                  ID2SYM(rb_intern("none")));
```

Comment:
You shouldn't need constants that hold symbols. The symbol itself should be a suitable constant for referring to a level. And if you get rid of the constants, you can just store the ID directly, like in the channel code (https://github.com/grpc/grpc/blob/master/src/ruby/ext/grpc/rb_channel.c#L377).